src/celpy/evaluation.py

```diff
@@ -293,6 +299,30 @@ def operator_in(item: Result, container: Result) -> Result:
     return result
 
 
+def _function_matches_re(text: str, pattern: str) -> Result:
+    try:
+        m = re.search(pattern, text)
+    except re.error as ex:
+        return CELEvalError("match error", ex.__class__, ex.args)
+
+    return celpy.celtypes.BoolType(m is not None)
+
+
+def _function_matches_re2(text: str, pattern: str) -> Result:
+    try:
+        m = re2.search(pattern, text)
+    except re2.error as ex:
+        return CELEvalError("match error", ex.__class__, ex.args)
+
+    return celpy.celtypes.BoolType(m is not None)
+
+
+def function_matches(text: str, pattern: str) -> Result:
```

Comment:
Can we hoist the `if` outside this function? 

```python
if _USE_RE2:
    def function_matches(...
        ... using _function_matches_re2()
else:
    def function_matches(...
        ... using _function_matches_re()
```

The idea is to avoid evaluating the `if` statement more than once at import time.